### PR TITLE
Stores the command line arguements within the task repository

### PR DESCRIPTION
**Pull request by mminella**
Resolves spring-cloud/spring-cloud-task#62

### CODE REVIEW

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java [hunk 1]
```diff
@@ -63,16 +66,20 @@
...
 	public TaskLifecycleListener(TaskRepository taskRepository,
-			TaskNameResolver taskNameResolver) {
+			TaskNameResolver taskNameResolver,
+			ApplicationArguments applicationArguments) {
 		Assert.notNull(taskRepository, "A taskRepository is required");
 		Assert.notNull(taskNameResolver, "A taskNameResolver is required");
 
 		this.taskRepository = taskRepository;
 		this.taskNameResolver = taskNameResolver;
+		this.arguments = applicationArguments;
```
Comment (cppwfs) [inline spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java]: nitpicky.  arguments should be applicationArguments.  Will change on merge.

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java [hunk 2]
```diff
@@ -60,41 +74,131 @@ public void tearDown() {
...
+		this.taskExplorer = context.getBean(TaskExplorer.class);
+
+		Sort sort = new Sort("id");
+
+		PageRequest request = new PageRequest(0, Integer.MAX_VALUE, sort);
+
+		Page<TaskExecution> taskExecutionsByName = this.taskExplorer.findTaskExecutionsByName("testTask",
+				request);
+
+		TaskExecution taskExecution = taskExecutionsByName.iterator().next();
```
Comment (cppwfs) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: Probably want to add an assertTrue for taskExecutionsByName.iterator().hasNext first before executing the next.

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java [hunk 3]
```diff
@@ -60,41 +74,131 @@ public void tearDown() {
...
+			return new SimpleApplicationArgs(args);
+		}
+	}
+
+	private static class SimpleApplicationArgs implements ApplicationArguments {
+
+		private Map<String, String> args;
+
+		public SimpleApplicationArgs(Map<String, String> args) {
+			this.args = args;
```
Comment (cppwfs) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: Probably need a Assert.notNull for args.

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java (hunk 2 above)
Comment (mminella) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: Given that it's a test, I am typically not as defensive but I can add it.

Comment (cppwfs) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: I can make the change on my merge.

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Merged + code review polishing

**Pull request closed without merging**